src/frame.test.js

```diff
@@ -0,0 +1,45 @@
+import Frame from './frame.js';
+
+let frame;
+
+describe("Frame", () => {
+  beforeEach(() => {
+    frame = new Frame();
+  });
+
+  test ("Strike if roll 10 in first go", () => {
+    frame.roll(10);
+    expect(frame.pinsLeft).toEqual(0);
+    expect(frame.score).toEqual(10);
+    expect(frame.result).toEqual("Strike");
+    expect(frame.rolls[0]).toEqual(10);
+  });
```

Comment:
In order to know if a test is testing state/implementation or behaviour, think of what should be accessible/visible outside the class. It's likely users of the Frame class only need to know the result and the score, and only need to input the rolls. That defines the interface of the class with the outside world, so that is hat your tests should focus on:
```
INPUT | OUTPUT
some rolls | expect score and result to be what I want them to be.
```